File: en-game/back-end/admin-service/src/main/java/com/endgame/adminservice/dto/product/ProductDTO.java
package com.endgame.adminservice.dto.product;

import com.endgame.adminservice.models.Color;
import com.endgame.adminservice.dto.image.ImageDTO;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.util.List;

@NoArgsConstructor
@Data
public class ProductDTO {
    private Long id;
    private String name;
    private BigDecimal newPrice = BigDecimal.ZERO;
    private BigDecimal oldPrice = BigDecimal.ZERO;
    private BigDecimal discount = BigDecimal.ZERO;
    private Long categoryId;
    private BigDecimal weight = BigDecimal.ZERO;
    private String description;
    private BigDecimal ratingCount = BigDecimal.ZERO;
    private BigDecimal ratingValue = BigDecimal.ZERO;
    private Long availibilityCount = 0L;
    private Long cartCount = 0L;
    private boolean active;
    private List<ImageDTO> images;
    private List<ImageDTO> imagesDeleted;
    private List<ImageDTO> imagesAdd;
    private ImageDTO imageMain;
    private List<Color> colors;
    private String size;
}

File: en-game/front-end-admin/src/app/main/dto/Product/ProductDTO.ts
import { ColorDTO } from "../Color/colorDTO";
import { ImageDTO } from "../image/imageDTO";

export class ProductDTO {
    id: number;
    categoryId: number;
    name: string;
    newPrice: number = 0;
    oldPrice: number = 0;
    discount: number = 0;
    ratingCount: number = 0;
    ratingValue: number = 0;
    availibilityCount: number = 0;
    cartCount: number = 0;
    weight: number = 0;
    size: string;
    description: string;
    active: boolean;
    // createdDate: string;
    images: ImageDTO[];
    imagesDeleted: ImageDTO[];
    imagesAdd: ImageDTO[];
    imageMain: ImageDTO;
    colors: ColorDTO[]
}